src/test/kotlin/ecommerce/repository/ProductRepositoryTest.kt

```diff
@@ -1,65 +1,73 @@
 package ecommerce.repository
 
-import ecommerce.model.Product
+import ecommerce.domain.Product
 import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
 import org.junit.jupiter.api.Test
 import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
 import org.springframework.boot.test.autoconfigure.jdbc.JdbcTest
 import org.springframework.context.annotation.Import
-import org.springframework.test.annotation.DirtiesContext
+import org.springframework.jdbc.core.JdbcTemplate
 
 @JdbcTest
 @Import(ProductRepository::class)
-@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@DisplayName("Product Repository Tests")
 class ProductRepositoryTest
     @Autowired
     constructor(
         private val productRepository: ProductRepository,
+        private val jdbcTemplate: JdbcTemplate,
     ) {
+        @BeforeEach
+        fun setUp() {
+            jdbcTemplate.execute("delete from products")
+        }
+
         @Test
         fun `it should count the products`() {
-            productRepository.create(Product(id = 1, "test", 20.0, "astr", 20))
-            productRepository.create(Product(id = 2, "test", 20.0, "astr", 20))
-            productRepository.create(Product(id = 3, "test", 20.0, "astr", 20))
+            val p1 = productRepository.create(Product(name = "test", price = 20.0, img = "http://astr", quantity = 20))
+            val p2 = productRepository.create(Product(name = "test2", price = 20.0, img = "http://astr", quantity = 20))
+            val p3 = productRepository.create(Product(name = "test3", price = 20.0, img = "http://astr", quantity = 20))
```

Comment:
I can feel that it must have been hard creating all these tests, but since we're getting close to perfection, why don't we give proper names to variables so that tests could be MORE PERFECT? 😄